locator/tablets.hh

```diff
@@ -359,8 +360,8 @@ public:
     /// \throws std::logic_error If the given id does not belong to this instance.
     dht::token_range get_token_range(tablet_id id) const;
 
-    /// Returns the primary replica for the tablet
-    host_id get_primary_replica(tablet_id id) const;
+    /// Returns the calculated primary replica map for the table
```

Comment:
This needs more explanation? What is a load map?

Reply:
BTW we could have kept the previous interface, with `tablet_map` calculating the load map on-demand and caching it for later uses.
IMO the old interface was more intuitive, but I don't have a strong opinion, this one is fine too.

Reply:
The problem with caching is when to reuse the cached map and when to evict it.
Assuming that the tablet_map and the topology are immutable, we can store it in the tablet_map.
But we need to make it is cleared when the tablet_map is cloned before we change it.

Reply:
Also, since for repair we may want a primary replica within a dc, we'll need to cache a primary replica per dc (plus a global one that can just be one of the per-dc primaries).  All in all, it seems overly complicated for a mechanism we currently use only in the control plane.

Reply:
I see that caching would make this complicated, so let's keep your current interface.

Reply:
kept with no caching in v11

Reply:
also improved the declaration comment (at least tried to make it clearer)